src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -968,7 +970,8 @@ void MacroAssembler::j(const address dest, Register temp) {
   if (is_simm21(distance) && ((distance % 2) == 0)) {
     Assembler::jal(x0, distance);
   } else {
-    assert(temp != noreg && temp != x0, "expecting a register");
+    assert(temp != x5 && temp != x1, "Register x5/x1 not used for jumps.");
```

Comment:
```suggestion
    assert(temp != noreg && temp != x0, "expecting a register");
    assert(temp != x5 && temp != x1, "Register x5/x1 should not be used for jumps.");
```
To keep the order consistent with https://github.com/openjdk/jdk/pull/21406/files#diff-7a5c3ed05b6f3f06ed1c59f5fc2a14ec566a6a5bd1d09606115767daa99115bdR943